model2vec/train/classifier.py

```diff
@@ -0,0 +1,293 @@
+            pred.append(torch.softmax(logits, dim=1).numpy())
+
+        return np.concatenate(pred)
+
+    def fit(
+        self,
+        X: list[str],
+        y: list[str],
+        learning_rate: float = 1e-3,
+        batch_size: int | None = None,
+        early_stopping_patience: int | None = 5,
+        test_size: float = 0.1,
+        device: str = "auto",
+    ) -> StaticModelForClassification:
+        """
+        Fit a model.
+
+        This function creates a Lightning Trainer object and fits the model to the data.
+        We use early stopping. After training, the weigths of the best model are loaded back into the model.
+
+        This function seeds everything with a seed of 42, so the results are reproducible.
+        It also splits the data into a train and validation set, again with a random seed.
+
+        :param X: The texts to train on.
+        :param y: The labels to train on.
+        :param learning_rate: The learning rate.
+        :param batch_size: The batch size.
+            If this is None, a good batch size is chosen automatically.
+        :param early_stopping_patience: The patience for early stopping.
+            If this is None, early stopping is disabled.
+        :param test_size: The test size for the train-test split.
+        :param device: The device to train on. If this is "auto", the device is chosen automatically.
+        :return: The fitted model.
+        """
+        pl.seed_everything(42)
+        logger.info("Re-initializing model.")
+        self._initialize(y)
+
+        train_texts, validation_texts, train_labels, validation_labels = self._train_test_split(
+            X, y, test_size=test_size
+        )
+
+        if batch_size is None:
+            batch_size = max(min(32, len(train_texts) // 10), 512)
+            logger.info("Batch size automatically set to %d.", batch_size)
+
+        logger.info("Preparing train dataset.")
+        train_dataset = self._prepare_dataset(train_texts, train_labels)
+        logger.info("Preparing validation dataset.")
+        val_dataset = self._prepare_dataset(validation_texts, validation_labels)
+
+        c = _ClassifierLightningModule(self, learning_rate=learning_rate)
+
+        n_train_batches = len(train_dataset) // batch_size
+        callbacks: list[Callback] = []
+        if early_stopping_patience is not None:
+            callback = EarlyStopping(monitor="val_accuracy", mode="max", patience=early_stopping_patience)
+            callbacks.append(callback)
+
+        # If the dataset is small, we check the validation set every epoch.
+        # If the dataset is large, we check the validation set every 250 batches.
+        if n_train_batches < 250:
+            val_check_interval = None
+            check_val_every_epoch = 1
+        else:
+            val_check_interval = max(250, 2 * len(val_dataset) // batch_size)
+            check_val_every_epoch = None
+
+        with TemporaryDirectory() as tempdir:
```

Comment:
This is to not create a random training directory when fitting?

Reply:
yep! We do need a training directory to be able to load the best checkpoint.